File: sample_java/src/main/java/com/memtrip/sample_java/HttpAudioResourceIntent.java
package com.memtrip.sample_java;

import android.content.Intent;

import com.memtrip.exoeasy.AudioResourceIntent;
import com.memtrip.exoeasy.NotificationInfo;

import org.jetbrains.annotations.NotNull;

public class HttpAudioResourceIntent extends AudioResourceIntent<HttpAudioResource> {

    private static final String EXTRA_PROPERTY = "EXTRA_PROPERTY";

    @NotNull
    @Override
    public Intent into(HttpAudioResource data, NotificationInfo notificationInfo, Intent intent) {
        intent.putExtra(EXTRA_PROPERTY, data.getExtraProperty());
        return super.into(data, notificationInfo, intent);
    }

    @NotNull
    @Override
    public HttpAudioResource get(Intent intent) {
        return new HttpAudioResource(
            intent.getStringExtra(HTTP_AUDIO_STREAM_URL),
            intent.getStringExtra(EXTRA_PROPERTY));
    }
}

File: sample/src/main/java/com/memtrip/exoeasy/sample/HttpAudioResourceIntent.kt
package com.memtrip.exoeasy.sample

import android.content.Intent

import com.memtrip.exoeasy.AudioResourceIntent
import com.memtrip.exoeasy.NotificationInfo

class HttpAudioResourceIntent : AudioResourceIntent<HttpAudioResource>() {

    override fun into(data: HttpAudioResource, notificationInfo: NotificationInfo, intent: Intent): Intent {
        intent.putExtra(EXTRA_PROPERTY, data.extraProperty)
        return super.into(data, notificationInfo, intent)
    }

    override fun get(intent: Intent): HttpAudioResource {
        return HttpAudioResource(
            intent.getStringExtra(HTTP_AUDIO_STREAM_URL),
            intent.getStringExtra(EXTRA_PROPERTY),
            intent.getStringExtra(HTTP_AUDIO_STREAM_USER_AGENT),
            intent.getBooleanExtra(HTTP_AUDIO_STREAM_TRACK_PROGRESS, true)
        )
    }

    companion object {
        const val EXTRA_PROPERTY = "EXTRA_PROPERTY"
    }
}